tests/services/portfolio.service.test.ts

```diff
@@ -0,0 +1,152 @@
+import portfolioService from '../../src/services/portfolio.service'
+import { IPortfolio } from '../../src/models/portfolio'
+import DatabaseManager from '../../src/mongo/databaseManager'
+import { populatePortfoliosInTestDb, cleanupDb } from '../utils/dbPopulation'
+import { testPortfolios } from '../utils/mockData'
+import { Types } from 'mongoose';
+import { MongoMemoryServer } from 'mongodb-memory-server';
+
+describe("Portfolio Service", () => {  
+    const mongod = new MongoMemoryServer();
+    let mongoDatabase: DatabaseManager;
+        
+    beforeAll(async (done) => {
+        const uri = await mongod.getConnectionString();
+        const dbName = await mongod.getDbName();
+
+        mongoDatabase = new DatabaseManager(
+            uri,
+            dbName
+        );
+
+        await mongoDatabase.connect();
+        await populatePortfoliosInTestDb(testPortfolios);
+        done();
+    });
+
+    afterAll(async (done) => {
+        await cleanupDb();
+        await mongoDatabase.disconnect();
+        await mongod.stop();
+        done();
+    });
+
+    it("gets a portfolio by id", async (done) => { 
+        let portfolio = await portfolioService.get(testPortfolios[0]._id.toHexString());
+        expect(portfolio).toEqual(testPortfolios[0]);
+        done();
+    });
+
+    it("gets a portfolio by one AND more fields", async (done) => {
+        let portfolio = await portfolioService.getByFields({
+            username: testPortfolios[0].name,
+            funds: []
+        });
+
+        expect(portfolio).toEqual(null);
+
+        portfolio = await portfolioService.getByFields({
+            name: testPortfolios[0].name,
+            funds: testPortfolios[0].funds
+        });
+
+        expect(portfolio).toEqual(testPortfolios[0]);
+        done();
+     });
+
+     it("gets all portfolios", async (done) => {
+        const portfolios = await portfolioService.getAll();
+        portfolios.forEach((portfolio: IPortfolio, index: number) => {
+            expect(portfolio).toEqual(testPortfolios[index]);
+        });
+        done();
+    });
+
+    it("gets a portfolio by one OR more fields", async (done) => { 
+        let portfolio = await portfolioService.getByEitherFields(
+            [
+                { name: testPortfolios[0].name },
+                { funds: [] }
+            ]);
+
+        expect(portfolio).toEqual(testPortfolios[0]);
+
+        portfolio = await portfolioService.getByEitherFields(
+            [
+                { name: 'randomname' },
+                { funds: testPortfolios[1].funds }
+            ]);
+
+        expect(portfolio).toEqual(testPortfolios[1]);
+        done();
```

Comment:
why do you have to specified you are done()??? so it doesnt continue with checks that are trivial for your test?